textalyze.js

```diff
@@ -43,9 +43,21 @@ function stringToCharacters(text) {
  */
 function sanitize(text) {
   if (typeof text === 'string' || text instanceof String) {
-    return text.toLowerCase();
+    return text.toLowerCase().replace(/[^a-zA-Z]+/g, '');
   }
-  return text;
+  return null;
+}
+
+/**
+ * Prints text frequencies histogram
+ * @param {Map} frequencies
+ */
+function printHistogram(frequencies) {
+  frequencies.forEach((freq, value) => {
+    const percentage = `[${freq < 0.1 ? ' ' : ''}${(freq * 100).toFixed(2)}%]`;
```

Comment:
Look at [String.prototype.padStart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart).  More broadly, any time you find yourself writing one-off code like this, double check whether there's something in the standard library that will do it (and if there is, use it).

Reply:
Er, wait, you use `padStart` below!  You could just do:

```javascript
const maxWidth = 200;
const freqStr = (freq * 100).toFixed(2).padLeft(5, ' ');
const bar = '='.repeat(freq * maxWidth);
console.log(`${value} [${freqStr}%] `${bar}`);
```

Reply:
The above code makes it absolutely clear what's going on at a glance.  That's what I'm optimizing for, here.

Reply:
Of course! I totally agree with you, thanks a lot!